application_sdk/application/metadata_extraction/generic.py

```diff
@@ -0,0 +1,186 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+from application_sdk.activities.metadata_extraction.generic import (
+    GenericMetadataExtractionActivities,
+)
+from application_sdk.application import BaseApplication
+from application_sdk.clients.generic import GenericClient
+from application_sdk.clients.utils import get_workflow_client
+from application_sdk.constants import MAX_CONCURRENT_ACTIVITIES
+from application_sdk.handlers.generic import GenericHandler
+from application_sdk.observability.decorators.observability_decorator import (
+    observability,
+)
+from application_sdk.observability.logger_adaptor import get_logger
+from application_sdk.observability.metrics_adaptor import get_metrics
+from application_sdk.observability.traces_adaptor import get_traces
+from application_sdk.server.fastapi import APIServer, HttpWorkflowTrigger
+from application_sdk.transformers import TransformerInterface
+from application_sdk.worker import Worker
+from application_sdk.workflows import WorkflowInterface
+
+logger = get_logger(__name__)
+metrics = get_metrics()
+traces = get_traces()
+
+
+class GenericMetadataExtractionApplication(BaseApplication):
```

Comment:
Check if this is actually required or will the BaseApplication class suffice
also override only the changing parts